src/viur/core/session.py

```diff
@@ -164,6 +166,29 @@ def reset(self) -> None:
         self.clear()
         self.changed = True
 
+    def __delitem__(self, key: str) -> None:
+        """
+            Removes a *key* from the session.
+            This key must exist.
+        """
+        del self[key]
+        self.changed = True
+
+    def __ior__(self, other: dict) -> t.Self:
+        """
+        Merges the contents of a dict into the session.
+        """
+        super().__ior__(other)
+        self.changed = True
+        return self
+
+    def update(self, other: dict) -> None:
+        """
+        Merges the contents of a dict into the session.
+        """
+        self |= other
+
```

Comment:
```suggestion
        self |= other
        self.changed = True
```
We change the session data here to.

Reply:
No, because `|=`-operator triggers __ior__, which sets the changed flag.